docs/development/agent-actor-system.md

```diff
@@ -0,0 +1,66 @@
+---
+layout: default
+nav_order: 1
+parent: Development
+---
+
+# Actor model and agent subsystems
+
+Zincati `agent` command provides a long-running background service which drives the OS through auto-updates.
+It comprises several logical subsystems which can run in parallel and are arranged following the [actor model][wiki-actors].
+
+The goal of this design is manifold:
+ * it allows splitting logical subsystems into separate failure domains.
+ * it models asynchronous operations and parallel components in an explicit way.
+ * it minimizes the amount of shared state and locks, minimizing the chances of deadlocks and concurrency bugs.
+
+[wiki-actors]: https://en.wikipedia.org/wiki/Actor_model
+
+## Actor system
+
+The core of the agent service is built on top of [Actix][actix], an asynchronous actor framework.
+The only exception is the initialization logic (i.e. CLI flags and configuration files parsing), which is performed in a synchronous way and only once at service startup.
+
+[actix]: https://github.com/actix/actix
+
+A general overview on such actor framework is provided by the [Actix book][actix-book], and further API details are covered by its [documentation](docs-rs-actix).
+There is also a companion web-framework called `actix-web`, which is however not relevant in this context.
+
+[actix-book]: https://actix.rs/book/actix/
+[docs-rs-actix]: https://docs.rs/actix
+
+The agent is split into several actors, which encapsulate state and behavior, as shown in the diagram below.
+They can run in parallel and communicate by exchanging messages.
```

Comment:
They can or they do?

Reply:
In general, they can. Some also run in parallel (which is useful in our context), but that depends on the specific arrangement.